src/snowflake/snowpark/session.py

```diff
@@ -3456,6 +3459,78 @@ def flatten(
         set_api_call_source(df, "Session.flatten")
         return df
 
+    def register_profiler(self, profiler: Profiler):
+        """Register a profiler to current session, all action are actually executed during this function"""
+        self.profiler = profiler
+        self.profiler.session = self
+        self.sql(f"show stages like '{profiler.stage}'").show()
+        self.sql(f"show stages like '{profiler.stage.split('.')[-1]}'").show()
```

Comment:
.show() will print things out, is this wanted?

Reply:
removed